Add script to update gradle deps related files

rmoreliovlabs opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Added script in order to update project dependencies

## Motivation and Context
In the project we keep a file with hashes of all dependencies we reference in the gradle file - ./gradle/verification-metadata.xml.

If new deps are added or existing ones are updated, we need to re-generate this XML file using this command:

`./gradlew --no-daemon --refresh-dependencies --write-verification-metadata sha256 assemble`

More info [here](https://docs.gradle.org/6.5/userguide/dependency_verification.html#sec:troubleshooting-verificaton).

As an improvement, we may have a shell script in the project, which would simplify this process. That script can run that long gradle command to re-generate the verification-metadata.xml file.

The script could simplify life of external contributors, as well as internal devs

Also we may update our wiki to mention about this capability.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)


* **Other information**:

Vovchyk reviewed (changes requested):
`update-deps.sh`? also I'd put it into the `scripts` folder, what do you think?

seems this command adds new entries of new dependencies to the metadata file, but also preserves older ones. Do you know if this command has an option to remove older / obsolete entries from the xml file?

rmoreliovlabs commented:
>

rmoreliovlabs commented:
> `update-deps.sh`? also I'd put it into the `scripts` folder, what do you think?
> 
> seems this command adds new entries of new dependencies to the metadata file, but also preserves older ones. Do you know if this command has an option to remove older / obsolete entries from the xml file?

Addressed comment but also found this: https://docs.gradle.org/current/userguide/dependency_verification.html#sec:verification-metadata-hygiene.

Let me know what you think about it

Vovchyk commented on scripts/update-verification-metadata:
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+cd ../
```
```suggestion

```

script is supposed to be executed in the project's root directory, so no need to `cd`

Vovchyk commented on scripts/update-verification-metadata:
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+cd ../
+rm gradle/verification-metadata.xml
+cp scripts/verification-metadata-template.xml gradle/verification-metadata.xml
+./gradlew --no-daemon --refresh-dependencies --write-verification-metadata sha256 assemble
```
```suggestion
set -e

# remove the verification-metadata.xml file with old deps (if any), and replace it with the template
if [ -f ./gradle/verification-metadata.xml ]; then
  rm ./gradle/verification-metadata.xml
  cp ./scripts/verification-metadata-template.xml ./gradle/verification-metadata.xml
fi

# generate verification-metadata.xml
./gradlew --no-daemon --refresh-dependencies --write-verification-metadata sha256 assemble
```

better to fail ( via `set -e` ) and exit, if for some reason file cannot be removed or the template is missing

Vovchyk reviewed (changes requested):
I'd also add the `.sh` extension to the script name to make it easily identifiable as a shell script

Vovchyk commented:
@rmoreliovlabs , I see you marked it as resolved. What are your thoughts on this?

> I'd also add the .sh extension to the script name to make it easily identifiable as a shell script

Also please make sure the pipeline is green

rmoreliovlabs commented:
> @rmoreliovlabs , I see you marked it as resolved. What are your thoughts on this?
> 
> > I'd also add the .sh extension to the script name to make it easily identifiable as a shell script
> 
> Also please make sure the pipeline is green

Sorry I forgot to push the changes

rmoreliovlabs commented:
pipeline:run

rmoreliovlabs commented:
pipeline:run

Vovchyk commented:
pipeline:run

Vovchyk commented:
pipeline:run

Vovchyk commented:
pipeline:run
